manifests/vrrp/sync_group.pp

```diff
@@ -3,38 +3,98 @@
+#
+# $notify_backup::         Define the notify backup script.
 #                          Default: undef.
 #
 # $notify_script_fault::   Define the notify fault script.
+#                          Depricated: Use $notify_fault
+#                          Default: undef.
+#
+# $notify_fault::          Define the notify fault script.
+#                          Default: undef.
+#
+# $notify_stop::           Executed when stopping vrrp
+#                          Default: undef.
+#
+# $notify_deleted::        Causes DELETED to be sent to notifies rather than the
+#                          default FAULT after a vrrp instance is deleted during
+#                          a reload. If a script is specified, that script will
+#                          be executed as well.
 #                          Default: undef.
 #
 # $notify_script::         Define the notify script.
+#                          Depricated: Use $notify
+#                          Default: undef.
+#
+# $notify::                Define the notify script.
+#                          Depricated: Use $notify
 #                          Default: undef.
 #
-# $notify_script_master_rx_lower_pri   Define the notify_master_rx_lower_pri script.
-#                          This is executed if a master receives an advert with
-#                          priority lower than the master's advert.
+# $notify_priority_changes:: Send FIFO notifies for vrrp priority changes
 #                          Default: undef.
 #
-# $smtp_alert::            Send email on status change (Boolean)
+# $smtp_alert::            Send email notification during state transition
 #                          Default: undef.
 #
+# $global_tracking::       DEPRECATED. Use track_interface, track_script and
+#                          track_file on vrrp_sync_groups instead.
+#                          Default: undef.
+#
+# $sync_group_tracking_weight:: Allow sync groups to use differing weights.
+#                          This probably WON'T WORK, but is a replacement for
+#                          global_tracking in case different weights were used
+#                          across different vrrp instances in the same sync group.
+#                          Default: undef.
 #
 define keepalived::vrrp::sync_group (
   $group,
-  Optional[Stdlib::Absolutepath] $notify_script_master_rx_lower_pri = undef,
-  $notify_script_master                                             = undef,
-  $notify_script_backup                                             = undef,
-  $notify_script_fault                                              = undef,
-  $notify_script                                                    = undef,
-  $smtp_alert                                                       = undef,
-  $nopreempt                                                        = undef,
-  Boolean $global_tracking                                          = false,
+  Variant[String, Array[String], Undef] $track_interface            = undef,
```

Comment:
the diff in this PR is quite huge? Do you think it's possible to cherry-pick the new datatypes for existing parameters into a single PR that we review first? Also I think 99% of those variables, if not all, should not allow empty strings. Could you update those to `String[1]` instead of `String`?